vm_server/accept/server.py

```diff
@@ -110,62 +148,151 @@ def execute_action(task_request, task_response):
     logging.debug("FAILED TO EXECUTE THE ACTION")
     task_response.status = Request_pb2.TaskResponse.FAILURE
     err = str(exception).encode(encoding)
+  status = execute.returncode
+  if status:
+    logging.debug("Execution was unsuccessful")
+    task_response.status = Request_pb2.TaskResponse.FAILURE
+  logging.debug("Process is running, force killing the process")
+  kill_process = subprocess.Popen("TASKKILL /F \
+                                  /PID {pid} /T".format(pid=execute.pid))
+  kill_process.communicate()
   if out is None:
     out = "".encode(encoding)
   if err is None:
     err = "".encode(encoding)
   try:
-    std_out = open(current_path + "\\output\\stdout.txt", "w")
+    std_out = open(current_path + OUTPUT_DIR + "stdout.txt", "w")
     std_out.write(out.decode(encoding))
     std_out.close()
-    std_err = open(current_path + "\\output\\stderr.txt", "w")
+    std_err = open(current_path + OUTPUT_DIR + "stderr.txt", "w")
     std_err.write(err.decode(encoding))
     std_err.close()
-    output_files = [name for name in os.listdir("..\\execute\\action\\output\\")
-                    if os.path.isfile("..\\execute\\action\\output\\" + name)]
+    output_files = [name for name in os.listdir(EXECUTE_ACTION_DIR + OUTPUT_DIR)
+                    if os.path.isfile(EXECUTE_ACTION_DIR + OUTPUT_DIR + name)]
     task_response.number_of_files = len(output_files)
     move_output(task_request, task_response)
   except Exception as exception:
-    logging.debug("Error writing in std out, stderr" + str(exception))
+    logging.debug("Error writing in stdout, stderr %s", str(exception))
     task_response.status = Request_pb2.TaskResponse.FAILURE
 
+def register_vm_address():
+  """Send request to master server to inform that VM is free"""
+  data = "http://" + VM_ADDRESS + ":" + str(PORT)
```

Comment:
data = "http://{}:{}".format(VM_ADDRESS, str(PORT)) is better than doing concatenation.

Reply:
Change has been made